packages/contracts/contracts/core/DAO.sol

```diff
@@ -10,96 +10,116 @@ import "@openzeppelin/contracts/utils/Address.sol";
 
 import "./erc1271/ERC1271.sol";
 import "./erc165/AdaptiveERC165.sol";
-import "./acl/ACL.sol";
+import "./permission/PermissionManager.sol";
 import "./IDAO.sol";
 
 /// @title The public interface of the Aragon DAO framework.
 /// @author Aragon Association - 2021
 /// @notice This contract is the entry point to the Aragon DAO framework and provides our users a simple and easy to use public interface.
-/// @dev Public API of the Aragon DAO framework
-contract DAO is IDAO, Initializable, UUPSUpgradeable, ACL, ERC1271, AdaptiveERC165 {
+/// @dev Public API of the Aragon DAO framework.
+contract DAO is IDAO, Initializable, UUPSUpgradeable, PermissionManager, ERC1271, AdaptiveERC165 {
     using SafeERC20 for ERC20;
     using Address for address;
 
-    // Roles
-    bytes32 public constant UPGRADE_ROLE = keccak256("UPGRADE_ROLE");
-    bytes32 public constant DAO_CONFIG_ROLE = keccak256("DAO_CONFIG_ROLE");
-    bytes32 public constant EXEC_ROLE = keccak256("EXEC_ROLE");
-    bytes32 public constant WITHDRAW_ROLE = keccak256("WITHDRAW_ROLE");
-    bytes32 public constant SET_SIGNATURE_VALIDATOR_ROLE =
-        keccak256("SET_SIGNATURE_VALIDATOR_ROLE");
-    bytes32 public constant MODIFY_TRUSTED_FORWARDER = keccak256("MODIFY_TRUSTED_FORWARDER");
+    /// @notice The ID of the permission required for the `_authorizeUpgrade` function.
+    bytes32 public constant UPGRADE_PERMISSION_ID = keccak256("UPGRADE_PERMISSION_ID");
 
+    /// @notice The ID of the permission required for the `setMetadata` function.
+    bytes32 public constant SET_METADATA_PERMISSION_ID = keccak256("SET_METADATA_PERMISSION_ID");
```

Comment:
Why permission ID? Wouldn't it be mode human centric to call it ROLE as it already was?

Reply:
I might agree with @brickpop . We should have a distinction between roles and permissions. I see them as separate things. currently, we work with roles, but in the future, ACL could also have permissions features. Happy to discuss this on the call..

Reply:
Permissions feels good then.
My remaining disturbance is about the _ID everywhere, but could live with that